test_verbosefilter.py

```diff
@@ -0,0 +1,108 @@
+# -*- python -*-
+
+"""
+Unit tests for VerboseFilter, a logging helper in adept_openstack.py
+"""
+
+import logging
+import sys
+from mock import Mock
+from unittest2 import TestCase, main
+
+sys.modules['virtualenv'] = Mock()      # Needed for Ed's Fedora system
+
+sys.path.insert(0, 'kommandir/bin')
+
+
+class MyHandler(logging.Handler):
+    """
+    Output handler for logging messages. No output; just push onto list.
+    """
+    def __init__(self, message_list):
+        logging.Handler.__init__(self)
+        self.message_list = message_list
+
+    def emit(self, record):
+        self.message_list.append(record.msg)
+
+
+class TestVerboseFilter(TestCase):
+    """
+    General test class. Defines only one test function, used by
+    test generator to create tests for all combinations of levels
+    and verbose flags.
+    """
+
+    def setUp(self):
+        import adept_openstack  # pylint: disable=E0401
+        self.a_o = adept_openstack
+        self.logger = logging.getLogger()
+        self.message_list = []
+
+    def _test_basic(self, level, verbose, expect):
+        # Grrr. The Logger thing is a global, and there doesn't seem to be
+        # a way to get a new one each time. For addHandler() and addFilter()
+        # to work, we need to delve into internals.
```

Comment:
Ya true...well, at least not without a whole lot of work.  Everything here is documented as python-2 (only) so this is fine since 2.7 is unlikely to ever change.